### PR TITLE
1699674 - Fix support for ES6 importing

### PR DESCRIPTION
**Pull request by brizental**

### CODE REVIEW

**Review by Dexterp37 — APPROVED** [review]
Thanks for the great work.

File: glean/package.json [hunk 1]
```diff
@@ -58,9 +58,9 @@
     "lint": "eslint . --ext .ts,.js,.json --max-warnings=0",
     "fix": "eslint . --ext .ts,.js,.json --fix",
     "build:cli": "tsc -p ./tsconfig/cli.json",
-    "build:webext:lib:esm": "tsc -p ./tsconfig/webext/esm.json",
+    "build:webext:lib:esm": "tsc -p ./tsconfig/webext/esm.json && echo '{\"type\": \"module\"}'> dist/webext/esm/package.json",
```
Comment (Dexterp37) [inline glean/package.json:61]: Please make sure to add a changelog entry fo rthis.

Comment (brizental) [reply glean/package.json]: Yeap, will do on Monday. Until then, would you kindly give this another review? There were a few changes to get tests and webpack to work again :D

File: glean/tests/core/metrics/labeled.spec.ts [hunk 2]
```diff
@@ -4,7 +4,6 @@
 
 import assert from "assert";
 import sinon from "sinon";
-import { JSONObject } from "../../../dist/webext/types/core/utils";
```
Comment (Dexterp37) [inline glean/tests/core/metrics/labeled.spec.ts:7]: Great catch!

File: glean/package.json [hunk 3]
```diff
@@ -52,15 +52,16 @@
...
+    "test:plugins": "npm run test:base -- \"tests/plugins/**/*.spec.ts\" --recursive",
+    "test:platform": "npm run build:test-webext && npm run test:base -- \"tests/platform/**/*.spec.ts\" --recursive --timeout 0",
+    "test:base": "node --experimental-modules --experimental-specifier-resolution=node --loader=ts-node/esm node_modules/mocha/lib/cli/cli.js",
     "build:test-webext": "cd tests/platform/utils/webext/sample/ && npm install && npm run build:xpi",
     "lint": "eslint . --ext .ts,.js,.json --max-warnings=0",
     "fix": "eslint . --ext .ts,.js,.json --fix",
     "build:cli": "tsc -p ./tsconfig/cli.json",
     "build:webext:lib:esm": "tsc -p ./tsconfig/webext/esm.json",
-    "build:webext:lib:cjs": "tsc -p ./tsconfig/webext/cjs.json",
+    "build:webext:lib:cjs": "tsc -p ./tsconfig/webext/cjs.json && echo '{\"type\": \"commonjs\"}'> dist/webext/esm/package.json",
```
Comment (Dexterp37) [inline glean/package.json:64]: Mh, given the configuration option, I would have expected this to happen automatically. Do we need to change [this](https://github.com/mozilla/glean.js/blob/6489e685ddcede776041dbea9af695b1d52c7a86/glean/tsconfig/webext/esm.json#L10) to commonjs?

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
Let's add a changelog entry :)

File: glean/package.json (hunk 3 above)
Comment (brizental) [reply glean/package.json:64]: Nope, the config you linked to is the ESM config, that needs `type: module`. Note that [before](https://github.com/mozilla/glean.js/pull/123/commits/93e9f37aae3e6e6739593d8e0ae0eb17e49ba33b) I had added the `type: module` to the ESM builds. For tests to work it is easier to leave the `type: module` declaration in the main package.json and override it with a `type: commonjs` in the commonjs config instead. Does this make sense?

Comment (brizental) [reply glean/package.json:64]: > given the configuration option, I would have expected this to happen automatically

How so?

Comment (Dexterp37) [reply glean/package.json:64]: NVM, you clarified it above, thanks.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,12 @@
 # Unreleased changes
 
+* [#123](https://github.com/mozilla/glean.js/pull/123): BUGFIX: Fix support for ES6 environments.
+  * Include `.js.` extensions in all local import statements.
```
Comment (Dexterp37) [inline CHANGELOG.md:4]:
```suggestion
  * Include `.js` extensions in all local import statements.
```

like this?

**Review by Dexterp37 — APPROVED** [review]
r+wc

**Pull request merged**